test/framework/integration_framework/integration_test_framework.cpp

```diff
@@ -102,26 +102,55 @@ namespace integration_framework {
         transaction_batch_factory_(
             std::make_shared<
                 shared_model::interface::TransactionBatchFactoryImpl>()),
-        deleter_(deleter) {}
+        cleanup_on_exit_(cleanup_on_exit) {}
```

Comment:
What was the reason to change this? 

Now it is not possible to pass custom deleter and invoke it over the instance.

Please change back or explain the change.

Reply:
It is very odd to initialize a class with a destructor code, and even more when it is used to call its internal methods. It breaks the encapsulation principle. The only way it was used was to either call the done() method, or do nothing, and the only purpose it was used was not to call the done() method in several cases. So the obvious fix is to tell the instance whether it should clean up on exit or no.